nBTC/sources/nbtc.move

```diff
@@ -172,59 +180,62 @@ public fun mint(
     let tx = tx::deserialize(&mut r);
 
     let tx_id = tx.tx_id();
-    let (mut amount, mut op_return) = verify_payment(
-        light_client,
-        height,
-        proof,
-        tx_index,
-        &tx,
-        contract.nbtc_bitcoin_pkh,
-    );
-
-    assert!(!contract.tx_ids.contains(tx_id), ETxAlreadyUsed);
-    assert!(amount > 0, EMintAmountIsZero);
 
-    let mut recipient: address = contract.get_fallback_addr();
-
-    if (op_return.is_some()) {
-        let msg = op_return.extract();
-        let mut msg_reader = reader::new(msg);
-        let flag = msg_reader.read_byte();
-        if (flag == 0x00) {
-            if (msg_reader.readable(32)) {
-                recipient = address::from_bytes(msg_reader.read(32));
-            };
-
-            // For flag=0x0 we expect only 32 bytes. If the stream is longer (more data), then
-            // the format is invalid, so moving recipient to fallback.
-            if (!msg_reader.end_stream()) {
-                recipient = contract.get_fallback_addr();
+    contract.nbtc_active_keys.keys().do_ref!(|pkh| {
+        let (mut amount, mut op_return) = verify_payment(
+            light_client,
+            height,
+            proof,
+            tx_index,
+            &tx,
+            *pkh,
+        );
```

Comment:
instead of looping let's do this:
- update the method to provide a key as an argument
- here we check if we know the key
- remove the loop